api/src/test/java/org/pmiops/workbench/api/ProfileControllerTest.java

```diff
@@ -593,6 +599,71 @@ public void testMe_removeAllInstitutionalAffiliations() throws Exception {
     assertThat(result.getInstitutionalAffiliations().size()).isEqualTo(0);
   }
 
+  @Test
+  public void updateContactEmail_forbidden() throws Exception {
+    createUser();
+    when(fireCloudService.isRequesterEnabledInFirecloud()).thenReturn(true);
+    user.setFirstSignInTime(new Timestamp(new Date().getTime()));
+    UpdateContactEmailRequest request = new UpdateContactEmailRequest();
+    request.setContactEmail(user.getContactEmail());
+    request.setUsername(user.getEmail());
+
+    ResponseEntity response = profileController.updateContactEmail(request);
+    assertThat(response.getStatusCode()).isEqualTo(HttpStatus.FORBIDDEN);
+  }
+
+  @Test
+  public void updateContactEmail_badRequest() throws Exception {
+    createUser();
+    when(fireCloudService.isRequesterEnabledInFirecloud()).thenReturn(true);
+    user.setFirstSignInTime(null);
+    UpdateContactEmailRequest request = new UpdateContactEmailRequest();
+    request.setContactEmail("bad email address *(SD&(*D&F&*(DS ");
+    request.setUsername(user.getEmail());
+
+    ResponseEntity response = profileController.updateContactEmail(request);
+    assertThat(response.getStatusCode()).isEqualTo(HttpStatus.BAD_REQUEST);
+  }
+
+  @Test
+  public void updateContactEmail_OK() throws Exception {
+    createUser();
+    when(fireCloudService.isRequesterEnabledInFirecloud()).thenReturn(true);
+    user.setFirstSignInTime(null);
+    UpdateContactEmailRequest request = new UpdateContactEmailRequest();
+    request.setContactEmail(user.getContactEmail());
+    request.setUsername(user.getEmail());
+
+    ResponseEntity response = profileController.updateContactEmail(request);
+    assertThat(response.getStatusCode()).isEqualTo(HttpStatus.NO_CONTENT);
+  }
+
+  @Test
+  public void resentWelcomeEmail_messagingException() throws Exception {
```

Comment:
Nit: Spelling (Should be `resend`? And below?)

Reply:
Good catch - that was on my todo list and I missed it.